src/celeritas/optical/OpticalPropertyData.hh

```diff
@@ -33,6 +48,7 @@ struct OpticalPropertyData
 
     Items<real_type> reals;
     OpticalMaterialItems<GenericGridData> refractive_index;
+    OpticalMaterialItems<ScintillationSpectra> scint_spectra;
```

Comment:
I'm not sure I understand why this is stored both in the optical property data and in the scintillation data. I think this struct should contain the optical properties from Geant4. I assume an application will set the properties needed for scintillation (like `SCINTILLATIONCOMPONENT[1,2,3]` and we will import those from Geant4 -- how do we plan to map those to the `ScintillationSpectra`?

Reply:
You are right! In principle, `ScintillationProcess` will/should prepare `ScintillationParams/Data` directly  from` MaterialView/Params`(?) if there is any scintillation material) so that data can be decoupled from other processes.  We should revisit a general approach how to handle optical materials and properties when optical processes are implemented along with optical material and property data.  I will remove `ScintillationSpectra` data from OpticalPropertyData and directly construct `ScintillationParams` for an initial test.

Reply:
Thanks @whokion, so if I understand correctly,  the Geant4 optical properties used for scintillation will more or less map directly to the `ScintillationSpectrum` fields. The only exception is the vector property `SCINTILLATIONCOMPONENT[1,2,3]`, the (Gaussian-like) scintillation emission spectrum tabulated for different photon energies, which in our data is replaced by the scalar properties `lambda_mean` and `lambda_sigma`, the parameters for sampling the wavelength from a Gaussian distribution. Regarding that:
 - Will the emission spectrum always be a Gaussian?
 - Since `SCINTILLATIONCOMPONENT` has energy (rather than wavelength) grid values, and ultimately we want to sample the energy, should we store `energy_mean` and `energy_sigma` instead?
 - Can we calculate the mean and stdev from the `SCINTILLATIONCOMPONENT`? If so, maybe this should be done in `ScintillationParams`, or at least as part of the test?
 - Or if not,  how do we get these parameters in an integrated Geant4 application? Do we require the user add new optical properties (which I guess can be done?) for the Gaussian distribution parameters and provide those instead of the tabulated data?

Reply:
- Gaussian should a good approximation as the scintillation wavelength is in reality a delta function (transition wavelength in the atom spectroscopy) with a (measurement) spread.
- As you know, we can always exchange energy vs wavelength, but the energy is a reciprocal Gaussian distribution as E~1/lambda if lambda is a Gaussian distribution which is much easier to sample randomly (than the reciprocal Gaussian). So, wavelength is a better choice for an input.
- Celeritias may provides an internal DB for constant properties (in this case, lambda_mean[] and lambda_sigma[] for known materials, but they should be in principle user inputs (at least, should support user defined constant properties).  So, this is a good item to consider how Celeritas supports optical properties (both constant and vector-like data) in a long run even there is no systematic solution in Geant4 either - one of our colleagues from MIT has proposed to support the better optical material and properties in Geant4 (like Geant4 defined optical materials and a complete set of associated optical properties) as he has a large collection of optical material properties, for which I suppose to be a technical assistant to integrate his DB into Geant4 (hope that this happens this year and also clear out all related licensing issues).